### ISSUE TITLE
Kyn Dropdown - Pre Selected Value from List Not Coming Up

### ISSUE THREAD

**Issue by guptaashray** [kyndryl-design-system/shidoka-applications#37]
This is the list I'm populating in the single select dropdown - 
[
    {
        "name": "Highly Critical",
        "content": "Highly Critical",
        "selected": false
    },
    {
        "name": "Somewhat critical",
        "content": "Somewhat critical",
        "selected": false
    },
    {
        "name": "Low",
        "content": "Low",
        "selected": true
    }
]

```
<kyn-dropdown [searchable]="false" size="xl" [placeholder]="selectCriticality" [disabled]="!service.businessCriticality.length" (on-change)="onInputDefnValuesCriticality($event)">
          <kyn-dropdown-option *ngFor="let option of businessCriticalitites" [value]="option.content [selected]="option.selected">{{option['content']}}</kyn-dropdown-option>
</kyn-dropdown>
```
The above is my HTML code for the same.

Now, the above values are rendered like this where I can see the selected value only when I open the dropdown in the second pic,  but does not appear when I don't open the dropdown as in the first pic.

<img width="838" alt="Screenshot 2023-11-30 at 10 42 04 PM" src="https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/0093fb4c-6dfb-4988-aa33-9b16755c8305">

<img width="805" alt="Screenshot 2023-11-30 at 10 42 31 PM" src="https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/46210bba-afb8-40b0-81dc-ff04232902fa">

Now if I open the dropdown and select any other option, it will appear.

<img width="796" alt="Screenshot 2023-11-30 at 10 45 19 PM" src="https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/3c8adf89-1ffa-4491-a95a-f6f4b3aeae19">

But, if I select the same option that was already selected, it will get deselected, but then if I again try to select that same option, I'm unable to see it appearing when the dropdown closes.
To see the already selected option appear, I need to open the dropdown and select any other option, and again select that option.

Expected - The already selected option should appear in the dropdown and the user should be able to deselect and select back the same option without selecting any other option.

https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/967cd69a-a844-45b3-ba5e-1436ca59bf3f

**Comment by brian-patrick-3** [comment]
This may be difficult to reproduce without being in the context of your specific application. The parent, `kyn-dropdown`, determines the initial selected value by looking for the `selected` `kyn-dropdown-option` child element on load, or whenever the slotted HTML changes. The fix for your previous issue was to reset the value when the slot content changes. Something about your app must be preventing it from finding the initial selected value.

**Comment by guptaashray** [comment]
businessCriticalities = 
[
  {
    "name": "Highly Critical",
    "content": "Highly Critical",
    "selected": false
  },
  {
    "name": "Somewhat critical",
    "content": "Somewhat critical",
    "selected": false
  },
  {
    "name": "Low",
    "content": "Low",
    "selected": true
  }
];

This is the list I'm populating, and using selected: true only for one option. I don't know, why it's not showing up initially!

```
<kyn-dropdown [searchable]="false" size="xl" [placeholder]="selectCriticality" [disabled]="!service.businessCriticality.length" (on-change)="onInputDefnValuesCriticality($event)">
          <kyn-dropdown-option *ngFor="let option of businessCriticalitites" [value]="option.content [selected]="option.selected">{{option['content']}}</kyn-dropdown-option>
</kyn-dropdown>
```

**Comment by guptaashray** [comment]
Issues Faced - 
1 - Dropdown options wipe out after selecting any option from the Dropdown
It is reproducible here itself - https://kyndryl-design-system.github.io/shidoka-applications/?path=/story/components-dropdown--data-driven-options

https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/e55e8542-ae48-4006-91d2-063f9b2be672

2 -  Pre-selected Values Not Coming Up in Single Select Dropdown and Count Coming Up for pre Selected Values in Multi Select Dropdown but tags not displaying the values unless and until the dropdown is Opened Once.

https://github.com/kyndryl-design-system/shidoka-applications/assets/43576436/51f5e95a-09e7-4e79-8275-eacf1cae3de2

**Comment by brian-patrick-3** [comment]
I seem to be unable to reproduce any of these issues in our Storybook on Windows Chrome or Firefox. @peterpham-kyndryl @maulik-thaker1 could you please give it a try as well?

**Comment by peterpham-kyndryl** [comment]
> I seem to be unable to reproduce any of these issues in our Storybook on Windows Chrome or Firefox. @peterpham-kyndryl @maulik-thaker1 could you please give it a try as well?

@brian-patrick-3 I can see the 1st issue on Mac + chrome. 
<img width="413" alt="Screenshot 2023-12-07 at 6 04 15 AM" src="https://github.com/kyndryl-design-system/shidoka-applications/assets/140734525/9e2d7c38-d847-4f65-974d-9a1b4f71ae6e">

Regarding the issue #2, I'm not seeing pre-selected value as an option in the controls panel on Storybook. And as in the video recorded. The select was being integrated, not on storybook. So I'm curious that on what framework he/she is consuming the component.

**Comment by brian-patrick-3** [comment]
Since I don't have a Mac, would you be able to make the fix for #1?

For pre-selected testing, just edit the stories locally to add `selected` attributes to some of the items. Can also add `multiple` to the dropdown in the data driven story.

They are using Angular I believe.

**Comment by guptaashray** [comment]
We are using the component on Angular 14 and the Shidoka version we are using is 1.0.4
